storage/storage.go

```diff
@@ -158,13 +161,33 @@ func NewClient(ctx context.Context, opts ...option.ClientOption) (*Client, error
 	}, nil
 }
 
+// newClientWithGRPC creates a new Storage client that initializes a gRPC-based client
+// for media upload and download operations.
+func newClientWithGRPC(ctx context.Context, opts ...option.ClientOption) (*Client, error) {
+	c, err := NewClient(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	g, err := gapic.NewClient(ctx, opts...)
```

Comment:
Good question. I thought they did, but turns out [they don't](https://github.com/googleapis/google-api-go-client/blob/896f71cdd1e1bb6aca6dcba186a0b401ad44bbee/transport/grpc/dial.go#L112).

Honestly, I think unsupported options should just be ignored...but maybe @codyoss has some thoughts? We will have to think of something here because the GAPIC will only be used for media operations and the HTTP client for metadata operations. I guess the factory method should have as separate `grpcOpts option.ClientOption` field, which kind of sucks and precludes making it variadic (only one of the two will be able to be variadic).

Reply:
Are there real examples where providing options here for both would cause issues? I know we do have some validation rules around not mixing some options: https://github.com/googleapis/google-api-go-client/blob/896f71cdd1e1bb6aca6dcba186a0b401ad44bbee/internal/settings.go#L115-L123

Reply:
@tritone do you know if there are some HTTP specific options folks use?

That is the code I was thinking about specifically, Cody. 

Do you think we could add some helper to google.golang.org/api/option that can split a slice of options into http & grpc options? Seems a little weird I think, but could make this a non-issue.

`func TransportOptions(opts ...ClientOption) (hOpts []ClientOption, gOpts []ClientOption)`

Those options that would work for both could be included in both.

Reply:
I think we could do something like that, but if we did I think it would be in `internaloption` as I don't think a consumer of this API would ever need to interact with that. 

I think there are some options that might overlap and not play nice. For example changing an endpoint. I think we need a way to change both two different values, right? If so, a new method that takes a config struct that can separate options might be in order instead.

Reply:
> I think there are some options that might overlap and not play nice. For example changing an endpoint. 

Yeah In my experience, use of `WithEndpoint` requires a scheme prefix, while gRPC doesn't allow that. That would be tricky.

> a new method that takes a config struct that can separate options might be in order instead.

Yeah, something like this if I'm following?

```go
// Could be in storage, could be in google.golang.org/api/option
type ClientOptions struct {
  // hOpts is the set of options used for a HTTP client.
  hOpts []option.ClientOption

  // gOpts is the set of options used for a gRPC client.
  gOpts []option.ClientOption
}


// In storage...could make it variadic to save the user passing in `nil`, but that would require merging all structs
func NewClientWithGRPC(ctx context.Context, opts *ClientOptions) (Client, error)
```

Reply:
Yup, it would be kinda annoying if you are not using auto-detection with ADC to pass auth options twice, but I think that is the only way to get it right 100% of the time.

Reply:
I've added an unexported `clientOptions` type for this.

Reply:
@broady could you weigh in?

Reply:
We've resolve this offline and are sticking with the struct.